tests/integration/construct_component_methods/go/main.go

```diff
@@ -72,6 +72,10 @@ func (o ComponentGetMessageResultOutput) Message() pulumi.StringOutput {
 	return o.ApplyT(func(v ComponentGetMessageResult) string { return v.Message }).(pulumi.StringOutput)
 }
 
+func (*Component) ElementType() reflect.Type {
+	return reflect.TypeOf((*Component)(nil))
+}
```

Comment:
The test fails after adding this (making `*Component` implement `pulumi.Input`), and succeeds after the fix in `context.go`.

Reply:
Input is such a scary interface. I'm developing an irrational fear that I'm not sure what happens when we declare something an input. I need to work on making this specific :)